src/sagemaker/session.py

```diff
@@ -5668,7 +5668,6 @@ def get_caller_identity_arn(self):
                 instance_name = metadata.get("ResourceName")
                 domain_id = metadata.get("DomainId")
                 user_profile_name = metadata.get("UserProfileName")
-                space_name = metadata.get("SpaceName")
```

Comment:
How did we ensure this is backwards compatible ?

Reply:
We should not be looking in default space settings for the execution role, so this part shouldn't be backwards compatible.